### PR TITLE
Feature/cornflow server migration

### PR DESCRIPTION
**Pull request by ggsdc**

### CODE REVIEW

File: cornflow-server/cornflow/shared/utils.py [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
 
-from flask_sqlalchemy import SQLAlchemy
-from flask_bcrypt import Bcrypt
+# from flask_sqlalchemy import SQLAlchemy
+# from flask_bcrypt import Bcrypt
 
-db = SQLAlchemy()
-bcrypt = Bcrypt()
+db = database
+bcrypt = password_crypt
```
Comment (marioncottard) [inline cornflow-server/cornflow/shared/utils.py:13]: I had the impression that in most files you imported database and password_crypt directly from cornflow_core, so I don't think they need to be defined here anymore

File: cornflow-server/cornflow/tests/custom_test_case.py [hunk 2]
```diff
@@ -205,6 +205,7 @@ def get_one_row(
             url, follow_redirects=True, headers=self.get_header_with_auth(token)
         )
 
+        print(row.json)
```
Comment (marioncottard) [inline cornflow-server/cornflow/tests/custom_test_case.py:208]: I think you can delete this

File: cornflow-server/cornflow/tests/unit/test_users.py [hunk 3]
```diff
@@ -146,6 +146,7 @@ def log_in(self, user):
 
     def get_user(self, user_asks, user_asked=None):
         data = {k: user_asks[k] for k in self.login_keys}
+        print(data)
```
Comment (marioncottard) [inline cornflow-server/cornflow/tests/unit/test_users.py:149]: You can delete this

File: cornflow-server/cornflow/tests/unit/test_users.py [hunk 4]
```diff
@@ -271,6 +272,7 @@ def test_admin_deletes_user(self):
         response = self.delete_user(self.admin, self.planner)
         self.assertEqual(200, response.status_code)
         response = self.get_user(self.admin, self.planner)
+        print(response.json)
```
Comment (marioncottard) [inline cornflow-server/cornflow/tests/unit/test_users.py:275]: This too

**Review by marioncottard — COMMENTED** [review]
Just a few things:

File: cornflow-server/cornflow/endpoints/meta_resource.py [hunk 5]
```diff
@@ -10,7 +10,8 @@
 # Import from internal modules
 from ..shared.authentication import Auth
 from ..shared.const import ALL_DEFAULT_ROLES
-from ..shared.exceptions import InvalidUsage, ObjectDoesNotExist, NoPermission
+from cornflow_core.exceptions import InvalidUsage, ObjectDoesNotExist, NoPermission
+from flask import g
 
 
 class MetaResource(Resource):
```
[the remark this answers is not in the document]
Comment (ggsdc) [reply cornflow-server/cornflow/endpoints/meta_resource.py:17]: The idea is to finally delete this class from here, it should not be needed

File: cornflow-server/cornflow/shared/utils.py (hunk 1 above)
Comment (ggsdc) [reply cornflow-server/cornflow/shared/utils.py:13]: It should not be needed, but I have not changed all the imports just yet

File: cornflow-server/cornflow/tests/custom_test_case.py (hunk 2 above)
Comment (ggsdc) [reply cornflow-server/cornflow/tests/custom_test_case.py:208]: You are completely right

File: cornflow-server/cornflow/tests/unit/test_users.py (hunk 3 above)
Comment (ggsdc) [reply cornflow-server/cornflow/tests/unit/test_users.py:149]: Ok

File: cornflow-server/cornflow/tests/unit/test_users.py (hunk 4 above)
Comment (ggsdc) [reply cornflow-server/cornflow/tests/unit/test_users.py:275]: Ok

File: cornflow-server/cornflow/endpoints/meta_resource.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (marioncottard) [reply cornflow-server/cornflow/endpoints/meta_resource.py]: Okay !

**Comment by marioncottard** [comment]
Tell me when you're done with the changes so I will review again

**Comment by ggsdc** [comment]
> Tell me when you're done with the changes so I will review again

I think the changes here are done, this should cover most of the generalization.

**Comment by marioncottard** [comment]
Are you done with the changes here ?

**Comment by ggsdc** [comment]
> Are you done with the changes here ?

I should be, but I am not 100% sure to make this changes just yet... I do not know why I am not 100% sure about them

**Review by marioncottard — APPROVED** [review]
Looks good to me ! Have you had any thought about what it is that makes you doubt ?

**Comment by ggsdc** [comment]
I'm going to finally merge the PR, I have merge with master, if all tests pass then I will merge this branch, although I think there is still things that can be improved on cornflow-tools and the generalization of cornflow-server

**Pull request merged**